examples/README-authz.md

```diff
@@ -0,0 +1,403 @@
+Moving the mouse pointer over it should reveal two additional realms - `Demo` and `Kafka-authz`.
+
+For this example we are interested in the `kafka-authz` realm.
+Selecting it will open the `Realm Settings` for the `kafka-authz` realm.
+Next to `Realm Settings` there are other sections we are interested in - `Groups`, `Roles`, `Clients` and `Users`.
+
+Under `Groups` we can see several groups that can be used to mark users as having some permissions.
+In Keycloak the groups can be stored in an LDAP identity provider.
+That makes it possible to make some user a member of some group - through custom LDAP server admin UI for example, which grants them some permissions on Kafka resources.
+
+Under `Users` click on `View all users` button to see two users defined - `alice` and `bob`. They are both members of some group. 
+In Keycloak the users can be stored in an LDAP identity provider.
+
+Under `Roles` we can see several realm roles which can be used to mark users or clients as having some permissions.
+Roles are a concept analogous to groups. One exception in Keycloak is that roles can not be stored in an LDAP identity provider.
+
+Under `Clients` we can see some additional clients configured - `kafka`, `kafka-cli`, `team-a-client`, `team-b-client`.
+The client with client id `kafka` is used by Kafka Brokers to perform the necessary OAuth2 communication for access token validation, 
+and to authenticate to other Kafka Broker instances using OAuth2 client authentication.
+This client also contains Authorization Services resource definitions, policies and authorization scopes used to perform authorization on Kafka Brokers.
+
+The client with client id `kafka-cli` is a public client that can be used by command line tools when authenticating with username and password to obtain an access token or a refresh token.
+
+Clients `team-a-client`, and `team-b-client` are confidential clients representing services with partial access to Kafka topics.
+
+The authorization configuration is defined in `kafka` client under `Authorization` tab.
+This tab becomes visible when `Authorization Enabled` is turned on under `Settings` tab.
+
+
+## Authorization Services - Resources, Authorization Scopes, Policies and Permissions
+
+If we take a look under `Resources` sub-tab of `Authorization` tab, we'll see the list of resource definitions.
+These are resource specifiers - patterns in a specific format, that are used to target policies to specific resources.
+The format is quite simple. For example:
+
+- `kafka-cluster:cluster-1,Topic:a_*`  ... targets only topics in kafka cluster 'cluster-1' with names starting with 'a_'
+
+If `kafka-cluster:XXX` segment is not present, the specifier targets any cluster.
+
+- `Group:x_*` ... targets all consumer groups on any cluster with names starting with 'x_'
+
+The possible resource types mirror the Kafka authorization model (Topic, Group, Cluster).
```

Comment:
The KAfka AuthZ model has a lot more than these 3. So can this work with all of them?

Reply:
Based on the [Kafka documentation](https://kafka.apache.org/documentation/#resources_in_kafka) the different resource types are: Topic, Group, Cluster, Transactionald, and DelegationToken. 

TransactionalId is supported, but I did not add support for DelegationToken since I imagine that's never used in conjunction with OAuth. If it is, it would require Authorizer to also work with non-oauth authentication, when there is no JWT token available.

Reply:
I'm not completely sure about the `DelegationToken`. But keep in mind this:
* There is only one authorizer per cluster
* There might be different interfaces with different auths and that is why we have the fallback option to the Simple authorizer.

So I think it is fine if we don't expect any `DelegationToken` rules in Keycloak. But there migth be usecases when we need to pass it to the fallback SImpleACLAuthorizer.

Reply:
I added all in the code, for completeness.